src/supertux/game_object_manager.cpp

```diff
@@ -213,8 +230,14 @@ GameObjectManager::flush_game_objects()
         }
       }
     }
+    s_needs_flushing_gameobjects_added = false;
+  }
+
+  if(s_needs_flushing_tilemaps)
+  {
+    update_tilemaps();
+    s_needs_flushing_tilemaps = false;
   }
-  update_tilemaps();
```

Comment:
Blocking `TileMap` updating to only when they are added is likely to cause issues, when a tilemap is dynamically set to solid, via the scripting function `set_solid()`.